plugins/hfp_ag_bluez4.c

```diff
@@ -99,14 +100,15 @@ static void hfp_ag_connect_cb(GIOChannel *io, GError *err, gpointer user_data)
 		return;
 	}
 
-	/* Pick the first voicecall capable modem */
-	modem = modems->data;
+	/* Pick the last voicecall capable modem */
+	last = g_list_last(modems);
```

Comment:
That cannot happen for the same reasons as stated for examples/emulator.c (see sim_state_watch()). Also, there is a senseless check right after this for a NULL modem, which seems wrong to me.